pages/profiles/[publicKey]/nfts.tsx

```diff
@@ -490,60 +489,63 @@ const ProfileNFTs: NextPage<WalletDependantPageProps> = (props) => {
-            <ListingFilter title={`All`} filterToCheck={ListingFilters.ALL} count={totalCount} />
-            <ListingFilter
-              title={`Listed`}
-              filterToCheck={ListingFilters.LISTED}
-              count={listedCount}
-            />
-            <ListingFilter
-              title={`Unlisted`}
-              filterToCheck={ListingFilters.UNLISTED}
-              count={unlistedCount}
-            />
-          </div>
-          <div className={`flex w-full lg:justify-end`}>
-            <TextInput2
-              id="owned-search"
-              label="owned search"
-              hideLabel
-              value={query}
-              onChange={(e) => setQuery(e.target.value)}
-              leadingIcon={
-                <FeatherIcon
-                  icon="search"
-                  aria-hidden="true"
-                  className={searchFocused ? 'text-white' : 'text-gray-500'}
-                />
-              }
-              placeholder="Search"
-              onFocus={() => setSearchFocused(true)}
-              onBlur={() => setSearchFocused(false)}
-            />
-            <GridSelector />
-          </div>
+    <>
+      <div className="sticky top-0 z-10 flex flex-col items-center gap-6 bg-gray-900 bg-opacity-80 py-4 backdrop-blur-sm lg:flex-row lg:justify-between lg:gap-4">
+        <div className={`flex w-full justify-start gap-4 lg:items-center`}>
+          <ListingFilter title={`All`} filterToCheck={ListingFilters.ALL} count={totalCount} />
+          <ListingFilter
+            title={`Listed`}
+            filterToCheck={ListingFilters.LISTED}
+            count={listedCount}
+          />
+          <ListingFilter
+            title={`Unlisted`}
+            filterToCheck={ListingFilters.UNLISTED}
+            count={unlistedCount}
+          />
         </div>
-        <NFTGrid
-          ctaVariant={`collected`}
-          hasMore={hasMore && filteredNfts.length > INITIAL_FETCH - 1}
-          onLoadMore={async (inView) => {
-            if (!inView || loading || filteredNfts.length <= 0) {
-              return;
+        <div className={`flex w-full lg:justify-end`}>
+          <TextInput2
+            id="owned-search"
+            label="owned search"
+            hideLabel
+            value={query}
+            onChange={(e) => setQuery(e.target.value)}
+            leadingIcon={
+              <FeatherIcon
+                icon="search"
+                aria-hidden="true"
+                className={searchFocused ? 'text-white' : 'text-gray-500'}
+              />
             }
-
+            placeholder="Search"
+            onFocus={() => setSearchFocused(true)}
+            onBlur={() => setSearchFocused(false)}
+          />
+          <GridSelector />
+        </div>
+      </div>
+      <NFTGrid
+        ctaVariant={`collected`}
+        hasMore={hasMore && filteredNfts.length > INITIAL_FETCH - 1}
+        onLoadMore={async (inView) => {
```

Comment:
would be awesome to extract this logic outside of the markup

Reply:
yesir